test/fetch/cookies.js

```diff
@@ -2,15 +2,27 @@
 
 const { once } = require('node:events')
 const { createServer } = require('node:http')
-const { test } = require('node:test')
+const { test, beforeEach } = require('node:test')
 const assert = require('node:assert')
 const { tspl } = require('@matteo.collina/tspl')
-const { Client, fetch, Headers } = require('../..')
+const { Client, fetch, Headers, Agent, setGlobalDispatcher } = require('../..')
 const { closeServerAsPromise } = require('../utils/node-http')
 const pem = require('https-pem')
 const { createSecureServer } = require('node:http2')
 const { closeClientAndServerAsPromise } = require('../utils/node-http')
 
+let agent
+beforeEach(() => {
```

Comment:
Since the test that needs this doesn't consume the body, won't this cause issues for people using fetch that don't consume the body? Especially the global fetch where Agent can't easily be changed.

Reply:
There is no need for this. I was trying to understand what was going on. I'll clean it up.